src/formatDate.js

```diff
@@ -8,7 +8,41 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[fromFormat.length - 1];
+  const separator2 = toFormat[toFormat.length - 1];
+
+  const curDate = date.split(separator);
+  const dateObj = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
```

Comment:
It's good that you're using a loop to assign date parts to `dateObj` based on `fromFormat`, but you should consider adding a check to ensure that the `curDate` array has the same number of elements as the `fromFormat` array minus one (to account for the separator). This will help catch cases where the input date does not match the expected format.